pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -222,6 +239,37 @@ func (f *framework) RunPrefilterPlugins(
 	return nil
 }
 
+// RunScorePlugins runs the set of configured scoring plugins. It returns a map that
+// stores for each scoring plugin name the corresponding NodeScoreList(s).
+// It also returns *Status, which is set to non-success if any of the plugins returns
+// a non-success status.
+func (f *framework) RunScorePlugins(pc *PluginContext, pod *v1.Pod, nodes []*v1.Node) (PluginToNodeScoreMap, *Status) {
+	pluginToNodeScoreMap := make(PluginToNodeScoreMap, len(f.scorePlugins))
+	for i := range f.scorePlugins {
+		pl := f.scorePlugins[i]
+		errCh := make(chan string, len(nodes))
+		plScores := make(NodeScoreList, len(nodes))
+		workqueue.ParallelizeUntil(context.TODO(), 16, len(nodes), func(index int) {
+			score, status := pl.Score(pc, pod, nodes[index].Name)
+			if !status.IsSuccess() {
+				errCh <- status.Message()
+			}
+			plScores[index] = NodeScore{
+				Host:  nodes[index].Name,
+				Score: score,
```

Comment:
We must multiply `score` by `pl.Weight`.

Reply:
where is the weight going to come from, add a method Weight() to the ScorePlugin interface?

Reply:
Done as per our offline discussion.